LiftSim/LiftSim/Floor.py

```diff
@@ -0,0 +1,128 @@
+# Imports
+from AjustableDataStore import AjustableDataStore as ads, UsageMethods as um
+from CustomDataTypes import *
+from CustomExeptions import *
+import numpy as np
+from Person import Person
+#import random.randint
+from  random import uniform
+
+class Floor(object):
+    """
+    Floor:
+        Description of class.
+
+        __init__(self, floorNum, weighting):
+            int floorNum - The floors number in the building.
+            float weighting - The weighting determining how likeley people are to want to arrive at the floor.
+
+        Public Atributes:
+            int FloorNumber - The floors number in the building.
+            float Weighting - The weighting determining how likeley people are to want to arrive at the floor.
+    """
+
+#-  Static Atributes
+    ArrivalMeans = None
+    FloorWeightings = None#TODO: create static initialisation function for these
+
+
+
+#-  Constructor
+    def __init__(self, floorNum):
+        """
+        Paramiters:
+            int floorNum - the floor number of the floor.
+        """
+        self.FloorNumber = floorNum
+        
+        self.__people = ads(um.Queue, Person)
+
+
+
+#-  Methods
+    def Update(self):
+        """
+        Updates the floor's state. Adds more people to the floor.
+        """
+        for i in count(randint(0, 3)):
+            self.__people.Push(Person(__selectDest(), self.FloorNumber, TickTimer.GetCurrentTick()))
+
+    def GetPeople(self, maxNumber):
+        """
+        Retrives a specified number of people at most from the floor.
+
+        Paramiters:
+            int maxNumber - The maximum number of people that can me returned.
+
+        Returns:
+            A list of Person objects: minimum length = 0, maximum length, "maxNumber"
+        """
+        if maxNumber >= self.__people.Count:
+            return self.__people.PopMany(maxNumber)
+        else:
+            return self.__people.PopMany(self.__people.Count)
+
+    def __selectDest(self):
+        statValues = Floor.FloorWeightings[self.FloorNumber, int(TickTimer.GetCurrentSecondsOfDay() / 3600)]
+
+        for prob in statValues:
+            total = total + float(prob)
+
+        chance = uniform(0, total)# Normal dist.?
+
+        for value in statValues:
+            iterator = iterator + float(value)
```

Comment:
Does this iterator need to be set to zero outside of this for loop? Otherwise it isn't defined so will throw an error.

Reply:
Proberbly. Its from the code I copyed in so ill take a look at it on monday.